File: backend/src/main/java/com/management/domain/entity/Tenant.java
package com.management.domain.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Entity
@Table(name = "tenants")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Tenant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "owner_user_id", nullable = false, updatable = false)
    private Long ownerUserId;

    @Column(name = "full_name", nullable = false, length = 255)
    private String fullName;

    @Column(length = 50)
    private String phone;

    @Column(name = "id_number", length = 50)
    private String idNumber;

    @Column(name = "id_type", length = 50)
    private String idType;

    @Column(length = 500)
    private String address;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @PrePersist
    protected void onCreate() {
        if (this.createdAt == null) {
            this.createdAt = Instant.now();
        }
    }
}

File: frontend/src/types/tenant.ts
export interface Tenant {
  id: number
  ownerUserId: number
  fullName: string
  phone: string | null
  idNumber: string | null
  idType: string | null
  address: string | null
  createdAt: string
}

export interface CreateTenantInput {
  fullName: string
  phone?: string
  idNumber?: string
  idType?: string
  address?: string
}

export interface UpdateTenantInput {
  fullName: string
  phone?: string
  idNumber?: string
  idType?: string
  address?: string
}
